### PR TITLE
add instructions for building and using the Docker environment

### PR DESCRIPTION
**Pull request by tukss**
These are some instructions for using the Docker container as a development environment. It uses our current Docker container on Docker Hub at the moment. In the future we might want to switch this to a newer build environment.

I chose to mount the home directory at `/root`. This has the advantage that `cd` goes to the right home directory and we can use the SSH keys from the normal home directory for git. I don't see a big problem in running the code as root inside the container, but if others prefer we could create an unprivileged  user inside the container. The downside would be that the users can't use `apt` directly to get new software.

### CODE REVIEW

File: doc/source/User_Guide/devel_environment.rst [hunk 1]
```diff
@@ -113,4 +118,48 @@ At this point, you can run "make install," and run the code using mpirun as you
...
+.. code-block:: bash
+
+    cd /root/path/to/Rayleigh
+    ./configure --with-fftw=/usr
+    make
+
+
+Building the container
+^^^^^^^^^^^^^^^^^^^^^^
+.. note:: This step purely optional. You only need to do this if you cannot pull the container from Docker Hub or you want to modify the Dockerfile.
```
Comment (gassmoeller) [inline doc/source/User_Guide/devel_environment.rst:159]: Maybe include that this has to happen *outside* of the container, and that you need to be in the Rayleigh directory.

**Review by gassmoeller — COMMENTED** [review]
I like it! I have one small questions, but everything else looks good.

File: doc/source/User_Guide/devel_environment.rst (hunk 1 above)
Comment (tukss) [reply doc/source/User_Guide/devel_environment.rst:159]: Good point. I added a commit with more details on building the container.

**Review by feathern — APPROVED** [review]
Thanks!

**Pull request merged**